folly/experimental/symbolizer/SignalHandler.cpp

```diff
@@ -361,8 +361,15 @@ void dumpSignalInfo(int signum, siginfo_t* siginfo) {
   printDec(getpid());
   print(" (pthread TID ");
   printHex((uint64_t)pthread_self());
+#if !defined(__FreeBSD__)
   print(") (linux TID ");
   printDec(syscall(__NR_gettid));
+#else
+  long tid;
+  thr_self(&tid);
```

Comment:
The functions called here need to be async-signal-safe.

I can't find info saying this is. It seems to be multi-thread-safe, but not async-signal-safe.

According to https://illumos.org/man/3C/thr_self
```

ATTRIBUTES

       See attributes(5) for descriptions of the following attributes:


       +---------------+-----------------+
       |ATTRIBUTE TYPE | ATTRIBUTE VALUE |
       +---------------+-----------------+
       |MT-Level       | MT-Safe         |
       +---------------+-----------------+
```

Is there any async-signal-safe way to implement this (eg. call a system call directly).